src/iceberg/manifest_entry.h

```diff
@@ -0,0 +1,256 @@
+struct ICEBERG_EXPORT DataFile {
+  /// Field id: 134
+  /// Type of content stored by the data file: data, equality deletes, or position
+  /// deletes (all v1 files are data files)
+  DataFileContent content;
+  /// Field id: 100
+  /// Full URI for the file with FS scheme
+  std::string file_path;
+  /// Field id: 101
+  /// File format type, avro, orc, parquet, or puffin
+  FileFormatType file_format;
+  /// Field id: 102
+  /// Partition data tuple, schema based on the partition spec output using partition
+  /// field ids for the struct field ids
+  /// TODO(zhjwpku): use StructLike to represent partition data tuple
+  std::map<std::string, std::any> partition;
+  /// Field id: 103
+  /// Number of records in this file, or the cardinality of a deletion vector
+  int64_t record_count = 0;
+  /// Field id: 104
+  /// Total file size in bytes
+  int64_t file_size_in_bytes = 0;
+  /// Field id: 108
+  /// Key field id: 117
+  /// Value field id: 118
+  /// Map from column id to the total size on disk of all regions that store the column.
+  /// Does not include bytes necessary to read other columns, like footers. Leave null for
+  /// row-oriented formats (Avro)
+  std::unordered_map<int32_t, int64_t> column_sizes;
+  /// Field id: 109
+  /// Key field id: 119
+  /// Value field id: 120
+  /// Map from column id to number of values in the column (including null and NaN values)
+  std::unordered_map<int32_t, int64_t> value_counts;
+  /// Field id: 110
+  /// Key field id: 121
+  /// Value field id: 122
+  /// Map from column id to number of null values in the column
+  std::unordered_map<int32_t, int64_t> null_value_counts;
+  /// Field id: 137
+  /// Key field id: 138
+  /// Value field id: 139
+  /// Map from column id to number of NaN values in the column
+  std::unordered_map<int32_t, int64_t> nan_value_counts;
+  /// Field id: 125
+  /// Key field id: 126
+  /// Value field id: 127
+  /// Map from column id to lower bound in the column serialized as binary.
+  /// Each value must be less than or equal to all non-null, non-NaN values in the column
+  /// for the file.
+  ///
+  /// Reference:
+  /// - [Binary single-value
+  /// serialization](https://iceberg.apache.org/spec/#binary-single-value-serialization)
+  std::unordered_map<int32_t, std::vector<uint8_t>> lower_bounds;
+  /// Field id: 128
+  /// Key field id: 129
+  /// Value field id: 130
+  /// Map from column id to upper bound in the column serialized as binary.
+  /// Each value must be greater than or equal to all non-null, non-Nan values in the
+  /// column for the file.
+  ///
+  /// Reference:
+  /// - [Binary single-value
+  /// serialization](https://iceberg.apache.org/spec/#binary-single-value-serialization)
+  std::unordered_map<int32_t, std::vector<uint8_t>> upper_bounds;
+  /// Field id: 131
+  /// Implementation-specific key metadata for encryption
+  std::optional<std::vector<uint8_t>> key_metadata;
```

Comment:
Is `std::vector<uint8_t>` sufficient?

Reply:
Yes, I believe an empty `key_metadata` is enough.

Reply:
Ok, I'll remove the optional part.